agenta-cli/agenta/client/client.py

```diff
@@ -1,112 +1,47 @@
-        timeout=600,
-    )
-    if response.status_code != 200:
-        error_message = response.json()
-        raise APIRequestError(
-            f"Request to get base failed with status code {response.status_code} and error message: {error_message}."
-        )
-    if len(response.json()) == 0:
-        raise APIRequestError(
-            f"Base with name {base_name} does not exist on the server."
-        )
-    else:
-        return response.json()[0]["base_id"]
-
-
-def get_app_by_name(app_name: str, host: str, api_key: str = None) -> str:
-    """Get app by its name on the server.
+        if "backend_host" not in self.config:
+            self.backend_host = "http://localhost"
+        else:
+            self.backend_host = self.config["backend_host"]
 
-    Args:
-        app_name (str): Name of the app
-        host (str): Hostname of the server
-        api_key (str): The API key to use for the request.
-    """
+        self.agenta_dir = Path.home() / ".agenta"
+        self.dir_config_file = self.agenta_dir / "config.toml"
+        self.dir_config = toml.load(self.dir_config_file)
+        self.api_key_for_fern = self.dir_config.get("api_key", None)
 
-    response = requests.get(
-        f"{host}/{BACKEND_URL_SUFFIX}/apps/?app_name={app_name}",
-        headers={"Authorization": api_key} if api_key is not None else None,
-        timeout=600,
-    )
-    if response.status_code != 200:
-        error_message = response.json()
-        raise APIRequestError(
-            f"Request to get app failed with status code {response.status_code} and error message: {error_message}."
+        self.api_client = client.AgentaApi(
+            base_url=f"{self.backend_host}/{self.backend_url_suffix}",
+            api_key=self.api_key_for_fern,
+            timeout=600,
         )
-    if len(response.json()) == 0:
-        raise APIRequestError(f"App with name {app_name} does not exist on the server.")
-    else:
-        return response.json()[0]["app_id"]  # only one app should exist for that name
-
 
-def create_new_app(app_name: str, host: str, api_key: str = None) -> str:
-    """Creates new app on the server.
-
-    Args:
-        app_name (str): Name of the app
-        host (str): Hostname of the server
-        api_key (str): The API key to use for the request.
-    """
 
-    response = requests.post(
-        f"{host}/{BACKEND_URL_SUFFIX}/apps/",
-        json={"app_name": app_name},
-        headers={"Authorization": api_key} if api_key is not None else None,
-        timeout=600,
-    )
-    if response.status_code != 200:
-        error_message = response.json()
-        raise APIRequestError(
-            f"Request to create new app failed with status code {response.status_code} and error message: {error_message}."
-        )
-    return response.json()["app_id"]
+class APIRequestError(Exception):
+    """Exception to be raised when an API request fails."""
 
 
 def add_variant_to_server(
```

Comment:
this logic should be another file and not the same as the wrapper. each module has its own responsability.